src/conv/invokers/impl_gemm.cpp

```diff
@@ -0,0 +1,147 @@
+
+InvokerFactory MakeImplGemmDataInvokerFactory(const ConvolutionContext& ctx)
+{
+    if(ctx.direction.IsForward())
+    {
+        return [](const std::vector<Kernel>& kernels) {
+            return [=](Handle& handle, const boost::any& primitive_parameters) {
+                const auto data_ctx = boost::any_cast<conv::DataInvokeParams>(primitive_parameters);
+                const auto& tensors = data_ctx.tensors;
+                handle.Run(kernels[0])(tensors.in, tensors.w, tensors.out);
+            };
+        };
+    }
+    else
+    {
+        const auto lowp_quant = ctx.conv_problem.GetConv().lowp_quant;
+
+        return [lowp_quant](const std::vector<Kernel>& kernels) {
+            return [=](Handle& handle, const boost::any& primitive_parameters) {
+                const auto data_ctx = boost::any_cast<conv::DataInvokeParams>(primitive_parameters);
+                const auto& tensors = data_ctx.tensors;
+                const auto& workSpace = data_ctx.workSpace;
+
+                // Miminum checks. Only check what is required to select
+                // proper invocation procedure & workspace sanity.
+                const auto kernel = handle.Run(kernels[0]);
+
+                float elapsed = 0;
+                if((tensors.outDesc.GetType() == miopenHalf ||
+                    tensors.outDesc.GetType() == miopenBFloat16) &&
+                   (kernel.GetName() == "gridwise_convolution_backward_data_implicit_gemm_v1r1_"
+                                        "xdlops_nchw_kcyx_nkhw" ||
+                    kernel.GetName() == "gridwise_convolution_backward_data_implicit_gemm_v1r1_"
+                                        "xdlops_gnchw_gkcyx_gnkhw" ||
+                    kernel.GetName() ==
+                        "gridwise_convolution_backward_data_implicit_gemm_v1r1_nchw_kcyx_nkhw"))
+                {
+                    float zero = 0.f;
+                    TensorDescriptor workspaceDesc(
+                        miopenFloat, tensors.outDesc.GetLengths(), tensors.outDesc.GetStrides());
+                    SetTensor(handle, workspaceDesc, workSpace, &zero);
+                    if(handle.IsProfilingEnabled())
+                        elapsed += handle.GetKernelTime();
+
+                    kernel(tensors.in, tensors.w, workSpace);
+                    if(handle.IsProfilingEnabled())
+                        elapsed += handle.GetKernelTime();
+
+                    CastTensor(handle,
+                               &lowp_quant,
+                               workspaceDesc,
+                               workSpace,
+                               tensors.outDesc,
+                               tensors.out,
+                               0,
+                               0);
+                    if(handle.IsProfilingEnabled())
+                        elapsed += handle.GetKernelTime();
+                }
+                else if((kernel.GetName() == "gridwise_convolution_implicit_gemm_v4_nchw_kc1x1_"
```

Comment:
kernel name is splitted into new line.

Better to keep it on the same line, so it's easy to grep the name

you can use "clang-format-off"